### DISCUSSION TITLE
[Feature] skywalking-ui just a front-end program, use the spring boot package, relying on too many jar increase in tar.gz' size

### DISCUSSION THREAD

**Discussion by davidricardo1026** [apache/skywalking#7752] [Ideas]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

The entire skywalking-webapp.jar size is 34.5m, but public folder after the package, only  963kb, skywalking Ui now just need an HTTP server, don't even need https, but spent 30m provides HTTP services
use light http server, so that almost 30m jar size can be reduced.If it is necessary
I can submit a PR, put executable JAR to the reduced to 3.3m, and then Public package, so that you can get the current 170m as a whole, down to 140m

### Use case

reduce skywalking-webapp.jar to 3.3m

### Related issues

none

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
To use Spring Gateway to host the website, is in order to make people easier to understand that. What container do you to propose? I think this kind of codebase change needs evaluation and PMC decision.

**Comment by davidricardo1026** [comment]
`Spring Gateway` is too heavy,act as just a http server,so may be a light http server is a better choice. and it still can make people easier to use it and experience it.use `Spring Gateway`,the jar is 34.5m,if a light http server,just 3-4m.

**Comment by wu-sheng** [comment]
I can get your point. Please provide a complete proposal, such as what new stack you want to use.

**Comment by davidricardo1026** [comment]
> I can get your point. Please provide a complete proposal, such as what new stack you want to use.

undertow-core can do,also simple,act as a http server,total jar size is only 3.3m,and not put the `public` to the jar,it will cheat people,let them thought `skywalking ui` is a java and spring project,but the turth is that `skywalking ui` is just a  front-end program and they can chose nginx to load banlance if they expect high availability.

**Comment by wu-sheng** [comment]
We set up the standard Gateway stack to provide users opportunity to choose more features if they like.
Also, according to Apache 2.0 License, anyone to set up their own webapp binary and image for more people in the community. 
I just say this from the official release perspective, and follow the tradition of the community. If you insist to change this, this would be longer term to make decision. I would like to recommend you try to set up a project to implement this first.

**Comment by davidricardo1026** [comment]
ok,thanks for your great work.
I will fork a branch to implement this first.

**Comment by wu-sheng** [comment]
I am moving this to discussion. Once we have more progress, such as you have those re-distribution and images, and after more people become to choose `webapp-slim` package, PMC could begin to discuss the replacement.
